cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -369,11 +369,11 @@ func StartControllers(s *options.CMServer, kubeconfig *restclient.Config, rootCl
 	// Find the list of namespaced resources via discovery that the namespace controller must manage
 	namespaceKubeClient := client("namespace-controller")
 	namespaceClientPool := dynamic.NewClientPool(restclient.AddUserAgent(kubeconfig, "namespace-controller"), restMapper, dynamic.LegacyAPIPathResolverFunc)
-	groupVersionResources, err := namespaceKubeClient.Discovery().ServerPreferredNamespacedResources()
+	deletableGroupVersionResources, err := namespaceKubeClient.Discovery().ServerPreferredNamespacedResourcesWithVerbs("delete")
```

Comment:
The serialization turns a `[]` into a null value in JSON. Am investigating why. Any idea? /cc @deads2k 

I had the variant with APIResource before. Didn't like it very much either. Also there are like 80 users of that interface in the code to change. And the interface is exported via client-go. So changing an interface is also not nice for 3rdparty users.